### PR TITLE
(PUP-9116 & PUP-3658) Update settings for Puppet 6

### PR DESCRIPTION
**Pull request by justinstoller**

### CODE REVIEW

File: lib/puppet/defaults.rb [hunk 1]
```diff
@@ -981,108 +981,78 @@ def self.default_vendormoduledir
...
     :cacrl => {
       :default => "$cadir/ca_crl.pem",
       :type => :file,
       :owner => "service",
       :group => "service",
       :mode => "0644",
-      :desc => "The certificate revocation list (CRL) for the CA. Will be used if present but otherwise ignored. This setting is deprecated and will be replaced by one in Puppet Server's configs in Puppet 6.",
-      :deprecated => :completely,
-    },
-    :caprivatedir => {
```
Comment (Magisus) [inline lib/puppet/defaults.rb]: We might want to make sure the PE isn't trying to use this or `capass`. I know they were asking me about them at some point and there was confusion around whether they were used or not. They may be setting them even though they are ignored. Not sure if that will cause issues.

**Comment by justinstoller** [comment]
Removed the CA setting removal as they also require removing the systems that use them (ie should be done it https://github.com/puppetlabs/puppet/pull/7021).

**Comment by Magisus** [comment]
Okay, I can update that there.

**Comment by justinstoller** [comment]
Not really? I'm under the impression that we're going to do a 6.x before we do our next 5.5.x so maybe we should cherry-pick back at that point if we haven't re-deprecated them by then?

**Comment by justinstoller** [comment]
However, I'm happy to open a PR against 5.5.x with the first commit if you think it best.

**Comment by Magisus** [comment]
Okay that works. I'm going to merge this.

**Pull request merged**